schemes/check_energy/dycore_energy_consistency_adjust.F90

```diff
@@ -0,0 +1,47 @@
+! MPAS and SE dynamical core specific
+! 1) scaling of temperature for enforcing energy consistency
+! 2) and to ensure correct computation of temperature dependent diagnostic tendencies, e.g., dtcore
+module dycore_energy_consistency_adjust
+  use ccpp_kinds, only: kind_phys
+  implicit none
+  private
+
+  public :: dycore_energy_consistency_adjust_run
+
+contains
+!> \section arg_table_dycore_energy_consistency_adjust_run Argument Table
+!! \htmlinclude arg_table_dycore_energy_consistency_adjust_run.html
+  subroutine dycore_energy_consistency_adjust_run( &
+      ncol, pver, &
+      energy_formula_dycore, &
+      scaling_dycore, &
+      temp_ini, &
+      t, &
+      tend_dtdt)
+
+    ! Dependency for energy formula definitions in SIMA
+    use cam_thermo_formula, only: ENERGY_FORMULA_DYCORE_SE, ENERGY_FORMULA_DYCORE_MPAS
+
+    ! Input arguments
+    integer,            intent(in)    :: ncol                  ! number of atmospheric columns
+    integer,            intent(in)    :: pver                  ! number of vertical layers
+    integer,            intent(in)    :: energy_formula_dycore ! total energy formulation dycore
+    real(kind_phys),    intent(in)    :: scaling_dycore(:,:)   ! scaling for conversion of   temperature increment [1]
+    real(kind_phys),    intent(in)    :: temp_ini(:,:)         ! initial temperature [K]
+
+    ! Input/output arguments
+    real(kind_phys),    intent(inout) :: T(:,:)                ! temperature [K]
+    real(kind_phys),    intent(inout) :: tend_dtdt(:,:)        ! model phys temperature tendency [K s-1]
+
+    if(energy_formula_dycore == ENERGY_FORMULA_DYCORE_MPAS .or. &
+       energy_formula_dycore == ENERGY_FORMULA_DYCORE_SE) then
+      T(:ncol,:) = temp_ini(:ncol,:) + &
+                   scaling_dycore(:ncol,:) * (T(:ncol,:) - temp_ini(:ncol,:))
+
+      tend_dtdt(:ncol,:) = scaling_dycore(:ncol,:) * tend_dtdt(:ncol,:)
+    endif
```

Comment:
Thanks for the suggestion! I understand the CCPP requirements but I have a couple concerns going that route:
* I am worried that applying this scaling of `T` as a tendency would obfuscate the actual formula. Making this scaling a tendency would look something like `((1 - temp_ini)*scaling_dycore + (scaling_dycore-1)*T))/dt`
* the scaling to `tend_dtdt` (`tendency_of_air_temperature_due_to_model_physics`) would still be modified directly, and since the tendency is applied later I would have to scale the above tendency by `1/scaling_dycore` as well.

For these reasons I am trying to argue for an exception here, but I'm happy to discuss this further!

Reply:
I am going to push back on this in that it isn't just a CCPP-requirement, but also a requirement if we ever want to do process-split physics, which is something that could very likely happen in CAM-SIMA as we start to do more weather-like simulations.  

I also personally disagree with the idea that it would make this scheme more complex. In fact I would argue that it would make it simpler.  This is because you can replace everything within the loop with just the following:

`tend_dtdt_local(:ncol,:) = (scaling_dycore(:ncol,:) - 1._r8) * tend_dtdt(:ncol,:)`

where `tend_dtdt_local` is `tendency_of_air_temperature` and `tend_dtdt` is `tendency_of_air_temperature_due_to_model_physics`.  Then calling the `apply_tendency_of_air_temperature` scheme right afterwards should result in the same final result (ignoring round-off changes).  I've spent time doing the math and believe I can algebraically prove that statement is true.

I realize that for CAM7 this isn't that important, but if we leave it as-is my concern is that it could produce subtle bugs for process-split physics suites that could be very difficult to track down.  So I would rather try and fix it now.

Also pinging @PeterHjortLauritzen in case he has any thoughts on this.

Reply:
Thanks for the suggestion! I stand corrected. I've implemented your suggestion and switched to using the tendency updater and verified answers are bit-for-bit in SIMA with `FKESSLER`, so this should be good to go.